File: javascript/ehtersExamples/examples/MyContract.js


class MyContract {

    // Constructor
    constructor(_name, _value) {
        this.name = String(_name);
        this.value = String(_value);
    };

    // Methods
    getName() {
        console.log(this.name);
        console.log(typeof(this.name));
        return (this.name);
    };

    getValue() {
        console.log(this.value);
        console.log(typeof(this.value));
        return this.value;
    };
};




// Deploy the Contract
const Contract = new MyContract('ERC20', 100);

// View Contract Data
Contract.getName();
Contract.getValue();

File: python/ethers_examples/MyContract.py

from dataclasses import dataclass


@dataclass
class MyContract:

    # default constructor
    name: str
    value: str

    # Methods
    def print_Name(self):
        _name = str(self.name)
        print(type(_name))
        print(_name)
        return _name

    def print_Value(self):
        _value = str(self.value)
        # _value = self.value
        print(type(_value))
        print(_value)
        return _value





# Deploy the Contract
Contract = MyContract('ERC20', 100)

# View Contract Data
Contract.print_Name()
Contract.print_Value()